Sources/NIO/SocketChannel.swift

```diff
@@ -598,14 +600,26 @@ class BaseSocketChannel<T: BaseSocket>: SelectableChannel, ChannelCore {
 
             // Call before triggering the close of the Channel.
             pipeline.fireChannelReadComplete0()
-            close0(error: err, mode: .all, promise: nil)
+            
+            if closeOnReadError(err) {
+                close0(error: err, mode: .all, promise: nil)
+            }
 
             return
         }
         pipeline.fireChannelReadComplete0()
         readIfNeeded0()
     }
 
+    /// Returns `true` if the `Channel` should be closed as result of the given `Error` which happened during `readFromSocket`.
+    ///
+    /// - parameters:
+    ///     - err: The `Error` which was thrown by `readFromSocket`.
+    /// - returns: `true` if the `Channel` should be closed, `false` otherwise.
+    fileprivate func closeOnReadError(_ err: Error) -> Bool {
```

Comment:
clearly a nit but `closeOnReadError` reads like it would close something. I'd rename it to `shouldCloseOnReadError`

Reply:
done